cli_tools/gce_vm_image_import/importer/validator.go

```diff
@@ -0,0 +1,57 @@
+//  Copyright 2020  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package importer
+
+import (
+	"fmt"
+	"google.golang.org/api/compute/v1"
+)
+
+// validator is the interface for performing validations.
+//
+// validate returns nil when passing, and non-nil when failing.
+type validator interface {
+	validate() error
+}
+
+func newPreValidator(args ImportArguments, client getImageClient) validator {
+	return validateImageNameAvailable{
+		project: args.Project,
+		name:    args.ImageName,
+		client:  client,
+	}
+}
+
+// validateImageNameAvailable is an importer.validator that
+// ensures the user's destination image name hasn't already
+// been used.
+type validateImageNameAvailable struct {
+	project, name string
+	client        getImageClient
+}
+
+func (v validateImageNameAvailable) validate() error {
+	// We ignore the error, with the assumption that if there's an error,
+	// then the image name may be available.
+	image, _ := v.client.GetImage(v.project, v.name)
+	if image != nil {
+		return fmt.Errorf("the image %q already exists. Please remove it, "+
+			"or pick a name that isn't already used", v.name)
```

Comment:
I don't think we should be recommending customers to remove existing images, as that sounds potentially dangerous and they could blame us if they lose any data. Stating image with the given name already exist should be enough. 

Also, it's a good idea to keep the existing error if any of the clients depend on a specific pattern. We can add more info, but we should keep existing error at least as an "intro" to the more verbose error.